### PR TITLE
Make appliesToType optional when appliesTo is contacts

### PR DESCRIPTION
**Pull request by kennsippell**
#208

### CODE REVIEW

File: test/nools/lib.tasks.spec.js [hunk 1]
```diff
@@ -44,6 +44,42 @@ describe('nools lib', function() {
...
+          targets: [],
+          tasks: [ aPersonBasedTask() ],
+        };
+        delete config.tasks[0].appliesToType;
+
+        // when
+        const { emitted } = runNoolsLib(config);
+
+        // then
+        console.log(JSON.stringify(emitted, null, 2));
```
Comment (dianabarsan) [inline test/nools/lib.tasks.spec.js]: I think you forgot to remove the logging :)

File: test/nools/lib.tasks.spec.js [hunk 2]
```diff
@@ -104,6 +140,42 @@ describe('nools lib', function() {
...
+          targets: [],
+          tasks: [ aReportBasedTask() ],
+        };
+        delete config.tasks[0].appliesToType;
+
+        // when
+        const { emitted } = runNoolsLib(config);
+
+        // then
+        console.log(JSON.stringify(emitted, null, 2));
```
Comment (dianabarsan) [inline test/nools/lib.tasks.spec.js]: Logging again.
It's weird, it looks like running `npm test` on Node > 8 will only log some form conversion  warnings and mocha reporting is completely silenced. Both on Travis and locally.

File: src/nools/lib.js [hunk 3]
```diff
@@ -84,8 +82,14 @@ if (tasks) {
 function emitTasksForSchedule(c, schedule, r) {
   var i;
 
-  if(r && schedule.appliesToType && schedule.appliesToType.indexOf(r.form) === -1) {
-    return;
+  if (schedule.appliesToType) {
+    var shouldApply = schedule.appliesTo === 'contacts' ?
+      c.contact && schedule.appliesToType.indexOf(c.contact.type) !== -1 :
+      r && schedule.appliesToType.indexOf(r.form) !== -1;
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/nools/lib.js]: You can have `appliesTo: 'scheduled_tasks'` and I don't fully understand all of those scenarios. So I'm not sure if this matters there, but I thought I'd keep as-is.

File: src/nools/lib.js [hunk 4]
```diff
@@ -71,9 +71,7 @@ if (tasks) {
         }
         break;
       case 'contacts':
-        if(c.contact && task.appliesToType.indexOf(c.contact.type) !== -1) {
-          emitTasksForSchedule(c, task);
-        }
+        emitTasksForSchedule(c, task);
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/nools/lib.js]: Agree. Good catch

**Comment by newtewt** [comment]
@kennsippell can you update to fix the conflicts? I'll re-review and merge.

**Comment by kennsippell** [comment]
@newtewt Done!

**Pull request merged**